crates/cast/src/cmd/call.rs

```diff
@@ -368,6 +379,20 @@ impl CallArgs {
 
         Ok(Some(state_overrides_builder.build()))
     }
+
+    /// Parse state overrides from command line arguments.
+    pub fn get_block_overrides(&self) -> eyre::Result<Option<BlockOverrides>> {
+        // Early return if no override set - <https://github.com/foundry-rs/foundry/issues/10705>
+        if [self.block_time.as_ref(), self.block_number.as_ref()].iter().all(Option::is_none) {
+            return Ok(None);
+        }
+
+        let block_overrides = BlockOverrides::default()
+            .with_number(U256::from(self.block_number.unwrap()))
+            .with_time(self.block_time.unwrap());
```

Comment:
oh I think this would currently be problematic because the is_none check only returns Ok(None) if all are none but this unwraps all.

we could do

```
let overrides = BlockOverrides {...}
if overrides.is_empty() return Ok(None)
```